### ISSUE TITLE
Majoras Mask Crash on Mac when entering the Stock Pott Inn for the first time

### ISSUE THREAD

**Issue by richard42** [mupen64plus/mupen64plus-user-issues#614]
Originally reported on Google Code with ID 615

```
iOS 10.9.2
64 bit MBP
Mupen64 - mupen64plus-bundle-osx-2.0.zip
Using the glide plugin (rice is too choppy)

When I enter the Stock Pot Inn, the game crashes - here's terminal info:
 __  __                         __   _  _   ____  _             
|  \/  |_   _ _ __   ___ _ __  / /_ | || | |  _ \| |_   _ ___ 
| |\/| | | | | '_ \ / _ \ '_ \| '_ \| || |_| |_) | | | | / __|  
| |  | | |_| | |_) |  __/ | | | (_) |__   _|  __/| | |_| \__ \  
|_|  |_|\__,_| .__/ \___|_| |_|\___/   |_| |_|   |_|\__,_|___/  
             |_|         http://code.google.com/p/mupen64plus/  
Mupen64Plus Console User-Interface Version 2.0.0

UI-Console: attached to core library 'Mupen64Plus Core' version 2.0.0
UI-Console:             Includes support for Dynamic Recompiler.
Core: Goodname: Legend of Zelda, The - Majora's Mask (U) [!]
Core: Name: ZELDA MAJORA'S MASK 
Core: MD5: 2A0A8ACB61538235BC1094D297FB6556
Core: CRC: 5354631c 3a2def0
Core: Imagetype: .z64 (native)
Core: Rom size: 33554432 bytes (or 32 Mb or 256 Megabits)
Core: Version: 144b
Core: Manufacturer: Nintendo
Core: Country: USA
UI-Console Status: Cheat codes disabled.
UI-Console: using Video plugin: 'Glide64mk2 Video Plugin' v2.0.0
UI-Console: using Audio plugin: 'Mupen64Plus SDL Audio Plugin' v2.0.0
Input Warning: get_hat_pos_by_name(): direction 'Right)"' unknown
Input Warning: get_hat_pos_by_name(): direction 'Left)"' unknown
Input Warning: get_hat_pos_by_name(): direction 'Down)"' unknown
Input Warning: get_hat_pos_by_name(): direction 'Up)"' unknown
UI-Console: using Input plugin: 'Mupen64Plus SDL Input Plugin' v2.0.0
UI-Console: using RSP plugin: 'Hacktarux/Azimer High-Level Emulation RSP Plugin' v2.0.0
Video: opening /Users/Rob-Old/Desktop/mupen64/mupen64plus.app/Contents/Resources/Glide64mk2.ini

INI_FindSection trying to find name for DEFAULT
Video: 3DNOW! detected.

Input: 1 SDL joysticks were found.
Input Warning: get_hat_pos_by_name(): direction 'Right)"' unknown
Input Warning: get_hat_pos_by_name(): direction 'Left)"' unknown
Input Warning: get_hat_pos_by_name(): direction 'Down)"' unknown
Input Warning: get_hat_pos_by_name(): direction 'Up)"' unknown
Input: N64 Controller #1: Using auto-config with SDL joystick 0 ('Generic   USB  Joystick
 ')
Input: 1 controller(s) found, 1 plugged in and usable in the emulator
Input: Mupen64Plus SDL Input Plugin version 2.0.0 initialized.
INI_FindSection trying to find name for DEFAULT
INI_FindSection trying to find name for ZELDA MAJORA'S MASK
Video: Using TEXUMA extension.

&ConfigOpenSection is 0x10fb3e0e8
(II) Setting video mode 1440x900...
Core: Setting video mode: 1440x900
NPOT extension used
use_fbo 1
Video: InitCombine() 
Video: extensions 
Video: initialized.
Video: 

Audio: Initializing SDL audio subsystem...
Cannot connect to server socket err = No such file or directory
Cannot connect to server socket
jack server is not running or cannot be started
Core: Starting R4300 emulator: Dynamic Recompiler
Core: R4300: starting 64-bit dynamic recompiler at: 0x1086d6d20
INI_FindSection trying to find name for UCODE
./run_glide.sh: line 1:   856 Segmentation fault: 11  ./mupen64plus.app/Contents/MacOS/mupen64plus
--corelib ./mupen64plus.app/Contents/MacOS/libmupen64plus.dylib --plugindir ./mupen64plus.app/Contents/MacOS
--gfx mupen64plus-video-glide64mk2 "$@"


I have the crash report too if that's helpful

```

Reported by `ImpromptuRob` on 2014-05-22 00:15:03

**Comment by richard42** [comment]
```
Does it still happen when compiling with following patch?
```

Reported by `conchur@web.de` on 2014-06-28 15:11:05

<hr>
- _Attachment: [replace_negative_int_shift.patch](https://storage.googleapis.com/google-code-attachments/mupen64plus/issue-615/comment-1/replace_negative_int_shift.patch)_

**Comment by richard42** [comment]
```
Or does this patch work better?
```

Reported by `conchur@web.de` on 2014-06-28 15:32:48

<hr>
- _Attachment: [replace_negative_int_shift2.patch](https://storage.googleapis.com/google-code-attachments/mupen64plus/issue-615/comment-2/replace_negative_int_shift2.patch)_

**Comment by richard42** [comment]
```
Can I ask a silly question - how to compile with a patch?
```

Reported by `ImpromptuRob` on 2014-06-28 18:45:45

**Comment by richard42** [comment]
```
Download the sources, apply the patch with http://linux.die.net/man/1/patch and then
compile it as usual. You can ask Richard@fascinationsoftware.com for more information.

Please test it:

 * new compiled glide64mk2 plugin without the patch
 * new compiled glide64mk2 plugin with the patch 1
 * new compiled glide64mk2 plugin with the patch 2
```

Reported by `conchur@web.de` on 2014-06-29 07:49:21

**Comment by richard42** [comment]
```
- current master (0e03ad142bfb4e236b674b218afc29f978a53fc1) crashes
- replace_negative_int_shift.patch crashes
- replace_negative_int_shift2.patch WORKS, can enter Stock Pot Inn without crash :)

Compiled from git with newest master branches on OSX 10.10.


```

Reported by `andrpoeh` on 2014-10-21 23:02:53

**Comment by richard42** [comment]
```
Tested this on the new 2.5 version using Glide64.  When you enter the Stock Pot Inn
from East Clock Town it immediately crashes.  If you use the deku flower and go in
through the second floor door you can get into the Inn but as soon as you go down the
stairs and enter the hallway on the left it will crash without fail.  

Running on OS X 10.10 
```

Reported by `keithnargi` on 2015-06-01 16:03:21

**Comment by richard42** [comment]
fixed after v2.5 in https://github.com/mupen64plus/mupen64plus-video-glide64mk2/commit/15f1f888422e248c6f640e587dffd12eeb48764b
